src/Avalonia.Controls/Calendar/CalendarExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// (c) Copyright Microsoft Corporation.
+// This source is subject to the Microsoft Public License (Ms-PL).
+// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
+// All other rights reserved.
+
+using System;
+using System.Collections.Generic;
+using Avalonia.Input;
+using System.Diagnostics;
+
+namespace Avalonia.Controls.Primitives
+{
+    internal static class CalendarExtensions
+    {
+
+        private static Dictionary<IAvaloniaObject, Dictionary<AvaloniaProperty, bool>> _suspendedHandlers = new Dictionary<IAvaloniaObject, Dictionary<AvaloniaProperty, bool>>();
```

Comment:
I don't think that this (and the related extension methods in `CalendarExtension` is needed. As far as I can tell, this was to get around the fact that Silverlight didn't have coercion on `DependencyProperty`s. WPF does, and Avalonia does (see [here](https://github.com/AvaloniaUI/Avalonia/blob/bd17aeffe20046c7bc35e5012e24cb9827727ab3/src/Avalonia.Controls/Grid.cs#L22) for example) so I think rather than checking the value is valid in the `OnXPropertyChanged` handler and setting it back if it's not, the relevant `AvaloniaProperty` should be given a `Validate` method. The suspension stuff can then be deleted from `CalendarExtensions`.

Does that make any sense?

Reply:
That makes sense and should be pretty easy to implement.